Updating Team Analytics, splitting out queue time and FRT (#375)
hellohaptik/docs_haptik: docs/agent-chat/analytics.md

title: Analytics
---

## Overview

![Overview](assets/overview.png)

1. **Avg. User rating**

    Average rating of all chats where the end user submitted a feedback and an agent was involved. Since this is a Team level view, we consider all agents in this team only, that is all chats where the agents from the chosen team took part.

2. **Total Number of agents**

    The value here gives us the number of agents, who received atleast a single chat in the Team. Only those agents who received any chat(s) are counted. Also note that an agent could be part of multiple teams. 

3. **Time saved of agents**

    All chats where an Haptik bot (gogo) took part, we add up the individual resolution time for all such chats, that would be the total amount of time, Haptik bot (gogo) was responding to the end user, instead of an human agent. 
    

> > This metric is calculated on the `business` level i.e. we show the "Time saved of agents" value totalled for all teams in a business (for the chosen Team). Since, most of our clients have one team per business, this apporach works. 

Author: @bhaveshpatelh i can remove this right?

Author: as in, we are currently showing time saved by gogo for a team only...

Reviewer: Yeah, This is not relevant anymore, we can remove it.

## Time Stats

![Time_stats](assets/time_stats.png)

1. **Average Agent First Response Time** (FRT)

    AFRT is the average of first response times taken by the agent to send the first response to end users after a chat was assigned to the agent. Only the chats which were initiated by users during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). As we check this from first user message timestamp to the first agent response timestamp.

> 2. **Average Queue Time** (RT)

Reviewer 2: I see in the Figma design that with queue time and without queue time will hold true for all Time stats. I think we should explain all these three matrices separately with queue time and without queue time.

Author: https://www.figma.com/file/iIKTe6BIMmokH3ELQRQxjl/Athena-Analytics?node-id=32%3A0

this is the design....

title: Analytics
---

> ## Overview

Author: all below metrics are based out of chat initiation time, within the selected time range

![Overview](assets/overview.png)

1. **Avg. User rating**

    Average rating of all chats where the end user submitted a feedback and an agent was involved. Since this is a Team level view, we consider all agents in this team only, that is all chats where the agents from the chosen team took part.

2. **Total Number of agents**

    The value here gives us the number of agents, who received atleast a single chat in the Team. Only those agents who received any chat(s) are counted. Also note that an agent could be part of multiple teams. 

3. **Time saved of agents**

    All chats where an Haptik bot (gogo) took part, we add up the individual resolution time for all such chats, that would be the total amount of time, Haptik bot (gogo) was responding to the end user, instead of an human agent.

## Time Stats

![Time_stats](assets/time_stats.png)

1. **Average Agent First Response Time** (FRT)

>     AFRT is the average of first response times taken by the agent to send the first response to end users after a chat was assigned to the agent. Only the chats which were initiated by users during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). As we check this from chat assignment to agent till his first response to the chat.

Author: ```suggestion
    Average agent first response time is the average of first response times taken by the agent to send the first response to end users after a chat was assigned to the agent. Only the chats which were initiated by users during the selected time range will be taken into account. 
> Note: This metric does not include Queue Time for the chats
```

2. **Average Queue Time** (RT)

    AQT is the average of total amount of time, chats have stayed within a queue. Chats remain in queue for the below mentioned reasons
    
    1. A fresh chat has been initiated for a team, and there are no free agent slots avaiable.

>     2. A chat has been re-assigned to an Agent who is currently at max concurrency.

Author: ```suggestion
    2. A chat has been re-assigned to an Agent who is currently at max concurrency.
    3. A chat was put in waiting state, when the user replied the chat got allocated back to the agent queue, but the agent was again at max concurrency.
```

>     In both of the scenarios mentioned above, the chat will remain in queue untill an agent receives the chat. 

Author: ```suggestion
    In all of the scenarios mentioned above, the chat will remain in queue untill an agent receives the chat. 
```

> > Number of free slots per agent depends on the number of chats currently assigned to the agent and chat concurrency set for the agent.

Author: ```suggestion
> Number of free slots per agent is the difference between chat concurrency set for the agent and the number of chats currently assigned to the agent.
```

3. **Average Response Time** (RT)

>     RT calculation is the average time taken by the agent to respond in a chat to all messages and not just the first message. Only the chats which had messages exchanged during the selected time range will be taken into account. In other words, average time taken by an agent to consecutively respond to a single or multiple user messages in cronological order.

Author: ```suggestion
    RT calculation is the average time taken by the agent to respond in a chat to all messages and not just the first message. Only the chats which were initiated during the selected time range will be taken into account. In other words, the average time taken by an agent to consecutively respond to a single or multiple user messages in chronological order.
```

    
4. **Average Agent Resolution Time**

>     This metric 'Average resolution time' should be the average time taken by the agent from the time when the chat was received till it was completed, re-assigned to marked as waiting. Only the chats that were completed during the selected time range will be taken into account. Chats may have been created anytime (inside and outside the selected time range). Agent Resolution time is simply calculated from the user’s first message sent timestamp till the last agent/user message timestamp in a chat, without queue time.

Author: ```suggestion
    This metric 'Average agent resolution time' should be the average time taken by the agent from the time when the chat was received till it was completed, re-assigned to, or marked as waiting. Only the chats that were initiated during the selected time range will be taken into account. Agent Resolution time is simply calculated from the user’s first message sent timestamp till the last agent/user message timestamp in a chat, without queue time.
```
